apps/ledger-live-desktop/src/renderer/modals/Receive/steps/StepReceiveStakingFlow.tsx

```diff
@@ -0,0 +1,180 @@
+      setAction(newAction);
+    }
+  }, [action, description, familyManageActions, id, t, title]);
+
+  const getTrackProperties = useCallback(() => {
+    return {
+      page: window.location.hash
+        .split("/")
+        .filter(e => e !== "#")
+        .join("/"),
+      modal: "receive",
+    };
+  }, []);
+
+  const openLink = useCallback(() => {
+    track("button_clicked", {
+      button: "learn_more",
+      ...getTrackProperties(),
+      link: supportLink,
+    });
+    openURL(supportLink);
+  }, [getTrackProperties, supportLink]);
+
+  const onChange = useCallback(() => {
+    const value = !doNotShowAgain;
+    global.localStorage.setItem(`${LOCAL_STORAGE_KEY_PREFIX}${id}`, value);
+    setDoNotShowAgain(value);
+    track("button_clicked", {
+      button: "not_show",
+      ...getTrackProperties(),
+      value,
+    });
+  }, [doNotShowAgain, getTrackProperties, id]);
+
+  return (
+    <Flex flexDirection="column" justifyContent="center" alignItems="center" px={10}>
+      <StakingIllustration />
+      <Text variant="large" mb={6} mt={12} fontWeight="semiBold" textAlign="center">
+        {title}
+      </Text>
+      <Text variant="paragraph" color="neutral.c70" textAlign="center">
+        {description}
+      </Text>
+      {supportLink && (
+        <Link Icon={Icons.ExternalLinkMedium} onClick={openLink} mt={7} mb={7} type={"color"}>
+          {t("receive.steps.staking.link")}
+        </Link>
+      )}
+      <Flex
+        p={6}
+        borderColor="neutral.c50"
+        borderRadius={2}
+        borderWidth={1}
+        borderStyle={"solid"}
+        onClick={onChange}
+      >
+        <CheckBox isChecked={doNotShowAgain} label={t("receive.steps.staking.notShow")} />
+      </Flex>
+    </Flex>
+  );
+};
+
+export const StepReceiveStakingFooter = (props: StepProps) => {
+  const { t } = useTranslation();
+  const [action, setAction] = useState({});
+  const { account, parentAccount, closeModal } = props;
+  const manage = perFamilyManageActions[account.currency.family];
+  const familyManageActions = manage && manage({ account, parentAccount });
+
+  useEffect(() => {
+    const manageList =
+      familyManageActions && familyManageActions.length > 0 ? familyManageActions : [];
+    const newAction = manageList && manageList.find(item => item.key === "Stake");
+    if (JSON.stringify(newAction) !== JSON.stringify(action)) {
+      setAction(newAction);
+    }
+  }, [action, familyManageActions]);
+
+  const getTrackProperties = useCallback(() => {
+    return {
+      page: window.location.hash
+        .split("/")
+        .filter(e => e !== "#")
+        .join("/"),
+      flow: "swap",
+      currency: account?.currency?.name,
+      provider: action?.provider?.name || "Ledger",
+      modal: "receive",
+      account,
+    };
+  }, [account, action?.provider?.name]);
+
+  const onStake = useCallback(() => {
+    if (action?.onClick) {
+      track("button_clicked", {
+        button: "swap",
```

Comment:
@sarneijim @xkleitz-ledger this is for stake but we are using swap, is that deliberate?